.github/scripts/workflow-stats.sh

```
@@ -0,0 +1,143 @@
+  # Increment total count
+  #echo "Workflow ${workflow_id} conclusion: $workflow_conclusion"
+  # Increment total count for success and failure
+  if [[ $workflow_conclusion == "failure" ]] || [[ $workflow_conclusion == "success" ]]; then
+    total_count=$((total_count + 1))
+  fi
+done
+
+# Function to calculate statistics with pruning
+calculate_stats() {
+  local array_name=$1
+  local times=($(eval echo "\${$array_name[@]}"))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)